types/async-retry/index.d.ts

```diff
@@ -29,7 +29,7 @@ import { WrapOptions } from "retry";
 declare function AsyncRetry<TRet>(fn: AsyncRetry.RetryFunction<TRet>, opts?: AsyncRetry.Options): Promise<TRet>;
 
 declare namespace AsyncRetry {
-    interface Options extends WrapOptions {
+    interface Options extends OperationOptions {
```

Comment:
I think extending `type OperationOptions` is wrong and the reason that [tests are failing](https://github.com/DefinitelyTyped/DefinitelyTyped/actions/runs/8945566321/job/24574768607?pr=69513#step:11:27). Previously this extended `interface WrapOptions`. 

I guess

```ts
type Options = OperationOptions & {
  onRetry?: ((e: Error, attempt: number) => any) | undefined;
}
```

is more correct?